lib/THC/THCReduce.cuh

```diff
@@ -11,24 +11,65 @@
 #include "THCTensorTypeUtils.cuh"
 #include "THCReduceApplyUtils.cuh"
 
-// Threads per thread block
-#define THC_NONCONTIG_REDUCE_BLOCK_SIZE 32 * 16
+#include <cstdio>
 
-template <typename IndexType>
-__device__ __forceinline__ IndexType getReduceNoncontigDimSliceIndex() {
-  // Each thread handles one slice
-  return getLinearBlockId<IndexType>() * THC_NONCONTIG_REDUCE_BLOCK_SIZE + threadIdx.x;
+#define LOCAL_MAX_BLOCK_SIZE 512
```

Comment:
It seems that this constant is used for shared mem size, but is not used when computing the block size. Is that ok?